analyses/cell-type-consensus/renv.lock

```diff
@@ -32,6 +32,50 @@
     "Version": "3.19"
   },
   "Packages": {
+    "AnnotationDbi": {
+      "Package": "AnnotationDbi",
+      "Version": "1.68.0",
+      "Source": "Bioconductor",
+      "Repository": "Bioconductor 3.20",
```

Comment:
Looks like the problem isn't just my local setup: https://github.com/AlexsLemonade/OpenScPCA-analysis/actions/runs/12354014198/job/34474381650?pr=936

Reply:
Yes, the fact that there is a mix of Bioc versions in the renv seems likely to be a problem. I'm quite surprised (but not totally shocked) that `renv` allowed that, but I am quite confident it is the problem. 

I think the move is most likely to revert (or delete) the `renv.lock` file, make sure the proper versions are set in `renv/settings.json` (they seem to be), then run `renv::hydrate()` to get a consistent set of packages.

Reply:
Just sticking an observation here: session info reports R version `4.4.2`

Reply:
> Just sticking an observation here: session info reports R version `4.4.2`

That's because I'm using R 4.4.2. That was the only version of R 4.4 I could find when I was setting up my new laptop that's available for arm (https://cran.r-project.org/bin/macosx/). I looked through the [old versions](https://cran.r-project.org/bin/macosx/base/) and couldn't find a version for arm, only Intel, so maybe I'm looking in the wrong place? 

Are we okay with using R 4.4.2 and BioC 3.20 for modules? The mismatching versions was because I had run the `create-analysis.py` script with the `--renv` flag which initiated the lock file. But when I actually created the project and started installing packages it used 3.20. I was able to fix the errors you were seeing by adjusting the lock file to have `R 4.4.2` and `Bioconductor 3.20`. I just want to be sure we are okay with keeping this version or should I try and change to R 4.4 and BioC 3.19? 


The error I'm seeing now looks like a [dependency error for igraph](https://r.igraph.org/articles/installation-troubleshooting.html#cannot-compile-igraph-from-sources-on-linux) and is unrelated to this.

Reply:
> I was able to fix the errors you were seeing by adjusting the lock file to have `R 4.4.2` and `Bioconductor 3.20`. I just want to be sure we are okay with keeping this version or should I try and change to R 4.4 and BioC 3.19?

This seems fine.

Reply:
> Are we okay with using R 4.4.2 and BioC 3.20 for modules? The mismatching versions was because I had run the `create-analysis.py` script with the `--renv` flag which initiated the lock file. But when I actually created the project and started installing packages it used 3.20. I was able to fix the errors you were seeing by adjusting the lock file to have `R 4.4.2` and `Bioconductor 3.20`. I just want to be sure we are okay with keeping this version or should I try and change to R 4.4 and BioC 3.19?

This like a bug we should fix... If our setup isn't forcing sticking with the defined Bioconductor version, we could end up with a lot of troubled down the road. 

(For the record, I am okay with R 4.4.2, but the difference between Bioc3.19 and 3.20 is potentially more significant)

Reply:
> This like a bug we should fix... If our setup isn't forcing sticking with the defined Bioconductor version, we could end up with a lot of troubled down the road.

This could have been a me problem. I probably didn't use `renv::restore()` when I first started writing code in this module, so I installed Bioconductor myself and it used the most recent version. So then I'm assuming when I installed any other packages it used the version of Bioconductor I had installed. If we want to use 3.19 then I can go back and delete everything and start again...